### PR TITLE
[13.3.][WIP]Tulip fixes

### PR DESCRIPTION
**Pull request by erikcas**
these changes give me a fully working twrp recovery.
Booting to system dies for me at the exit of bootanimation

### CODE REVIEW

**Comment by erikcas** [comment]
should help to fix https://github.com/sonyxperiadev/kernel/issues/824

**Comment by erikcas** [comment]
the revert in c82757d59b9d7dd8743bb541d73c70c0bb8d1453 is trivial.
The revert brings life into @bartcubbins device.

TODO (imo) is we should rather fix what breaks boot, then revert the whole commit

@kholk any thoughts?

**Comment by bartcubbins** [comment]
@erikcas offline charging works when the device is connected to the charger. when connected to a PC device goes into flashmode.

**Comment by kholk** [comment]
Yes. Does the device just die with no reason? 
Complete lockup and reboot at android init end is something that I've experienced a lot of times while playing with "exotic" things, like different kernel versions and different kernel (os)  architectures. 

The first reason for those lockups can be a bad programming of the cpu low power manager blocks: fast off-mode states of the cores, like core retention etc. are managed by "magic" commands to that block, which are also managed by the underlying TZOS running in the permanently running TZ block of the SoC. 

This usually happens only with devices running the SoC in high security mode because of bugs in either TZOS, kernel driver or both. 

In short, there's something you can try: go to Android init rcs, you will find a line that writes 0 to the "lpm_disable" file located in sysfs. Remove that line. That will disable LP states, you will have no deep sleep and high power consumption in this state but, if it works, then we know what's the problem. 

Let me know how it goes!

**Comment by erikcas** [comment]
@bartcubbins you are able to test @kholk  suggesstion? Unf I still cannot find my beloved tulip......

**Comment by bartcubbins** [comment]
@erikcas @kholk  yes, I'll test it.

**Comment by bartcubbins** [comment]
@kholk I understand correctly?
https://github.com/bartcubbins/device-sony-kanuti/commit/a6136a663086cdd2bf0a3c38aaf6b73908999814

**Comment by bartcubbins** [comment]
@erikcas @kholk I tested that using smb1360 from a current branch, the device is turned on, but when the cable is disconnected - reboot. And device is not booting without charger or PC connection.

**Comment by kholk** [comment]
Yes you got it right and I confused the filename. 
So the thing didn't work.... That's good more or less, it means that lpm is fine. That would have been a real pain to fix. 

Strange issue, anyway.

**Comment by pablomh** [comment]
Isn't this the same behavior that @jerpelea was having with his tulip even before this commit: it booted only attached to a USB port?

**Comment by pablomh** [comment]
I've been able to finally boot my tulip (e2333) into 7.1.1 using this kernel. Thanks @erikcas @bartcubbins!!

**Comment by jerpelea** [comment]
[   74.622624] wcnss_notif_cb: wcnss notification event: 6
[   74.627992] ------------[ cut here ]------------
[   74.632369] WARNING: at fs/sysfs/dir.c:530 sysfs_add_one+0x98/0xb4()
[   74.638745] sysfs: cannot create duplicate filename '/devices/soc.0/qcom,rpm-smd.14/rpm-regulator-ldoa3.98/regulator-l3-corner-ao.100/regulator/regulator.13/a000000.qcom,wcnss-wlan-qcom,pronto-vddmx'
[   74.656461] CPU: 1 PID: 1 Comm: init Not tainted 3.10.84-g2125ec6-01651-g88d831f #1
[   74.657685] type=1400 audit(1225688.309:6): avc: denied { read } for pid=378 comm="ueventd" name="wcnss.mdt" dev="mmcblk0p18" ino=27 scontext=u:r:ueventd:s0 tcontext=u:object_r:vfat:s0 tclass=file permissive=1
[   74.657943] type=1400 audit(1225688.309:7): avc: denied { open } for pid=378 comm="ueventd" path="/firmware/image/wcnss.mdt" dev="mmcblk0p18" ino=27 scontext=u:r:ueventd:s0 tcontext=u:object_r:vfat:s0 tclass=file permissive=1
[   74.702638] Call trace:
[   74.705021] [<ffffffc000206d98>] dump_backtrace+0x0/0x23c
[   74.710539] [<ffffffc000206fe4>] show_stack+0x10/0x1c
[   74.715475] [<ffffffc000e10a54>] dump_stack+0x1c/0x28
[   74.716694] type=1400 audit(1225688.309:8): avc: denied { getattr } for pid=378 comm="ueventd" path="/firmware/image/wcnss.mdt" dev="mmcblk0p18" ino=27 scontext=u:r:ueventd:s0 tcontext=u:object_r:vfat:s0 tclass=file permissive=1
[   74.740771] [<ffffffc00021ca4c>] warn_slowpath_common+0x70/0x9c
[   74.746596] [<ffffffc00021cb00>] warn_slowpath_fmt+0x5c/0x80
[   74.752394] [<ffffffc0003691e0>] sysfs_add_one+0x94/0xb4
[   74.757563] [<ffffffc000369c50>] sysfs_do_create_link_sd+0xfc/0x1c0
[   74.763783] [<ffffffc000369f10>] sysfs_create_link+0x2c/0x38
[   74.769492] [<ffffffc000555764>] create_regulator+0xc4/0x224
[   74.775069] [<ffffffc000555998>] _regulator_get+0xd4/0x154
[   74.780559] [<ffffffc000555b80>] regulator_get+0xc/0x18
[   74.785747] [<ffffffc00068bd64>] wcnss_vregs_on+0x8c/0x1f4
[   74.791246] [<ffffffc00068c4f4>] wcnss_wlan_power+0xa8/0x218
[   74.796857] [<ffffffc000689fc4>] wcnss_notif_cb+0x74/0x1f0
[   74.802357] [<ffffffc000243394>] notifier_call_chain+0x44/0x80
[   74.808219] [<ffffffc000243808>] __srcu_notifier_call_chain+0x48/0x70
[   74.814566] [<ffffffc000243840>] srcu_notifier_call_chain+0x10/0x1c
[   74.820954] [<ffffffc00054a9d8>] subsys_notif_queue_notification+0x1c/0x30
[   74.827723] [<ffffffc00054b5c0>] notify_each_subsys_device.isra.12+0x188/0x1c8
[   74.834891] [<ffffffc00054c574>] notify_proxy_vote+0x84/0xac
[   74.840704] [<ffffffc00053cd1c>] pil_boot+0x1b4/0x56c
[   74.845567] [<ffffffc00053d6c4>] subsys_powerup+0x20/0x2c
[   74.851061] [<ffffffc00054beb0>] __subsystem_get+0xfc/0x200
[   74.856506] [<ffffffc00054bfc0>] subsystem_get+0xc/0x18
[   74.861826] [<ffffffc00068af1c>] wcnss_node_open+0xc8c/0xf80
[   74.867388] [<ffffffc00058b388>] misc_open+0xb8/0x10c
[   74.872378] [<ffffffc00030a77c>] chrdev_open+0x11c/0x168
[   74.877809] [<ffffffc000304440>] do_dentry_open+0x180/0x234
[   74.883228] [<ffffffc000305400>] finish_open+0x34/0x54
[   74.888462] [<ffffffc00031295c>] do_last.isra.37+0x860/0xa38
[   74.893993] [<ffffffc000312bfc>] path_openat.isra.38+0xc8/0x420
[   74.900007] [<ffffffc000312f90>] do_filp_open+0x3c/0xa8
[   74.905103] [<ffffffc000305894>] do_sys_open+0x138/0x1e8
[   74.910509] [<ffffffc000305978>] SyS_openat+0xc/0x18
[   74.915344] ---[ end trace e22e83f0e8948cd5 ]---
[   74.920011] 8916_l3_corner_ao: could not add device link a000000.qcom,wcnss-wlan err -17
[   74.924306] warning: `qseecomd' uses 32-bit capabilities (legacy support in use)
[   74.935477] 8916_l3_corner_ao: Failed to create debugfs directory
[   74.941591] ------------[ cut here ]------------
[   74.946082] WARNING: at fs/sysfs/dir.c:530 sysfs_add_one+0x98/0xb4()
[   74.952476] sysfs: cannot create duplicate filename '/devices/soc.0/qcom,rpm-smd.14/rpm-regulator-smpa2.86/regulator-s2-corner.87/regulator/regulator.5/a000000.qcom,wcnss-wlan-qcom,pronto-vddcx'
[   74.969721] CPU: 1 PID: 1 Comm: init Tainted: G        W    3.10.84-g2125ec6-01651-g88d831f #1
[   74.978345] Call trace:
[   74.980748] [<ffffffc000206d98>] dump_backtrace+0x0/0x23c
[   74.986095] [<ffffffc000206fe4>] show_stack+0x10/0x1c
[   74.991151] [<ffffffc000e10a54>] dump_stack+0x1c/0x28
[   74.996164] [<ffffffc00021ca4c>] warn_slowpath_common+0x70/0x9c
[   75.002101] [<ffffffc00021cb00>] warn_slowpath_fmt+0x5c/0x80
[   75.007759] [<ffffffc0003691e0>] sysfs_add_one+0x94/0xb4
[   75.013003] [<ffffffc000369c50>] sysfs_do_create_link_sd+0xfc/0x1c0
[   75.019283] [<ffffffc000369f10>] sysfs_create_link+0x2c/0x38
[   75.024897] [<ffffffc000555764>] create_regulator+0xc4/0x224
[   75.030565] [<ffffffc000555998>] _regulator_get+0xd4/0x154
[   75.036006] [<ffffffc000555b80>] regulator_get+0xc/0x18
[   75.041258] [<ffffffc00068bd64>] wcnss_vregs_on+0x8c/0x1f4
[   75.046683] [<ffffffc00068c4f4>] wcnss_wlan_power+0xa8/0x218
[   75.052357] [<ffffffc000689fc4>] wcnss_notif_cb+0x74/0x1f0
[   75.057824] [<ffffffc000243394>] notifier_call_chain+0x44/0x80
[   75.063610] [<ffffffc000243808>] __srcu_notifier_call_chain+0x48/0x70
[   75.070071] [<ffffffc000243840>] srcu_notifier_call_chain+0x10/0x1c
[   75.076284] [<ffffffc00054a9d8>] subsys_notif_queue_notification+0x1c/0x30
[   75.083171] [<ffffffc00054b5c0>] notify_each_subsys_device.isra.12+0x188/0x1c8
[   75.090375] [<ffffffc00054c574>] notify_proxy_vote+0x84/0xac
[   75.095989] [<ffffffc00053cd1c>] pil_boot+0x1b4/0x56c
[   75.101053] [<ffffffc00053d6c4>] subsys_powerup+0x20/0x2c
[   75.106406] [<ffffffc00054beb0>] __subsystem_get+0xfc/0x200
[   75.111990] [<ffffffc00054bfc0>] subsystem_get+0xc/0x18
[   75.117196] [<ffffffc00068af1c>] wcnss_node_open+0xc8c/0xf80
[   75.122813] [<ffffffc00058b388>] misc_open+0xb8/0x10c
[   75.127876] [<ffffffc00030a77c>] chrdev_open+0x11c/0x168
[   75.133142] [<ffffffc000304440>] do_dentry_open+0x180/0x234
[   75.138724] [<ffffffc000305400>] finish_open+0x34/0x54
[   75.143819] [<ffffffc00031295c>] do_last.isra.37+0x860/0xa38
[   75.149489] [<ffffffc000312bfc>] path_openat.isra.38+0xc8/0x420
[   75.155364] [<ffffffc000312f90>] do_filp_open+0x3c/0xa8
[   75.161660] [<ffffffc000305894>] do_sys_open+0x138/0x1e8
[   75.165957] [<ffffffc000305978>] SyS_openat+0xc/0x18
[   75.170928] ---[ end trace e22e83f0e8948cd6 ]---
[   75.175488] 8916_s2_corner: could not add device link a000000.qcom,wcnss-wlan err -17
[   75.176640] apr_tal:Modem Is Up
[   75.178497] diag: In diag_send_feature_mask_update, control channel is not open, p: 0, 0000000000000000
[   75.195866] 8916_s2_corner: Failed to create debugfs directory
[   75.201724] ------------[ cut here ]------------
[   75.206209] WARNING: at fs/sysfs/dir.c:530 sysfs_add_one+0x98/0xb4()
[   75.212571] sysfs: cannot create duplicate filename '/devices/soc.0/qcom,rpm-smd.14/rpm-regulator-ldoa7.108/regulator-l7.109/regulator/regulator.18/a000000.qcom,wcnss-wlan-qcom,pronto-vddpx'
[   75.229517] CPU: 0 PID: 1 Comm: init Tainted: G        W    3.10.84-g2125ec6-01651-g88d831f #1
[   75.238094] Call trace:
[   75.240496] [<ffffffc000206d98>] dump_backtrace+0x0/0x23c
[   75.245874] [<ffffffc000206fe4>] show_stack+0x10/0x1c
[   75.251035] [<ffffffc000e10a54>] dump_stack+0x1c/0x28
[   75.255945] [<ffffffc00021ca4c>] warn_slowpath_common+0x70/0x9c
[   75.261888] [<ffffffc00021cb00>] warn_slowpath_fmt+0x5c/0x80
[   75.267515] [<ffffffc0003691e0>] sysfs_add_one+0x94/0xb4
[   75.272783] [<ffffffc000369c50>] sysfs_do_create_link_sd+0xfc/0x1c0
[   75.279057] [<ffffffc000369f10>] sysfs_create_link+0x2c/0x38
[   75.284677] [<ffffffc000555764>] create_regulator+0xc4/0x224
[   75.290351] [<ffffffc000555998>] _regulator_get+0xd4/0x154
[   75.295788] [<ffffffc000555b80>] regulator_get+0xc/0x18
[   75.301025] [<ffffffc00068bd64>] wcnss_vregs_on+0x8c/0x1f4
[   75.306465] [<ffffffc00068c4f4>] wcnss_wlan_power+0xa8/0x218
[   75.312131] [<ffffffc000689fc4>] wcnss_notif_cb+0x74/0x1f0
[   75.317599] [<ffffffc000243394>] notifier_call_chain+0x44/0x80
[   75.323391] [<ffffffc000243808>] __srcu_notifier_call_chain+0x48/0x70
[   75.329838] [<ffffffc000243840>] srcu_notifier_call_chain+0x10/0x1c
[   75.336065] [<ffffffc00054a9d8>] subsys_notif_queue_notification+0x1c/0x30
[   75.342948] [<ffffffc00054b5c0>] notify_each_subsys_device.isra.12+0x188/0x1c8
[   75.350154] [<ffffffc00054c574>] notify_proxy_vote+0x84/0xac
[   75.355771] [<ffffffc00053cd1c>] pil_boot+0x1b4/0x56c
[   75.360829] [<ffffffc00053d6c4>] subsys_powerup+0x20/0x2c
[   75.366187] [<ffffffc00054beb0>] __subsystem_get+0xfc/0x200
[   75.371767] [<ffffffc00054bfc0>] subsystem_get+0xc/0x18
[   75.376950] [<ffffffc00068af1c>] wcnss_node_open+0xc8c/0xf80
[   75.382617] [<ffffffc00058b388>] misc_open+0xb8/0x10c
[   75.387652] [<ffffffc00030a77c>] chrdev_open+0x11c/0x168
[   75.392925] [<ffffffc000304440>] do_dentry_open+0x180/0x234
[   75.398521] [<ffffffc000305400>] finish_open+0x34/0x54
[   75.403602] [<ffffffc00031295c>] do_last.isra.37+0x860/0xa38
[   75.409276] [<ffffffc000312bfc>] path_openat.isra.38+0xc8/0x420
[   75.415145] [<ffffffc000312f90>] do_filp_open+0x3c/0xa8
[   75.420382] [<ffffffc000305894>] do_sys_open+0x138/0x1e8
[   75.425648] [<ffffffc000305978>] SyS_openat+0xc/0x18
[   75.430620] ---[ end trace e22e83f0e8948cd7 ]---
[   75.435212] 8916_l7: could not add device link a000000.qcom,wcnss-wlan err -17

**Comment by jerpelea** [comment]
please remove 18db6db6f86d28ee602f4c86e40553bda9991657 we use pn547

**Comment by pablomh** [comment]
@jerpelea, I would leave that for further improvement. Unless we want a "With this commit Everything Works (TM)" franken-commit

**Comment by jerpelea** [comment]
we use a different nfc hw so there is no point merging that commit

**Comment by pablomh** [comment]
@bartcubbins, how is it working your NFC code?

**Comment by pablomh** [comment]
@jerpelea, after re-reading your comment, I agree that there's no point in adding (not modifying) that code.

**Comment by bartcubbins** [comment]
@pablomh give me 30min and I create PR

**Comment by erikcas** [comment]
@jerpelea the wcnss logging you posted above, afaik that has been present from the beginnen on all devices using prima. I once contacted @kholk to fix that, he kindly requested me (in his very own and well respected way) not to touch it if I remember well

**Comment by bartcubbins** [comment]
@erikcas please resolve conflicts after this pr https://github.com/sonyxperiadev/kernel/pull/1165

**Comment by pablomh** [comment]
I can confirm that WiFi is working.
